src/xo-app/vm/tab-disks.js

```diff
@@ -516,11 +643,49 @@ export default class TabDisks extends Component {
       isAdmin || (resourceSet == null && isVmAdmin)
   )
 
+  individualActions = [
+    {
+      handler: this._migrateVdi,
+      icon: 'vdi-migrate',
+      label: _('vdiMigrate'),
+    },
+    {
+      disabled: (vdi, userData) => {
+        const vbd = find(userData.vbdsByVdi[vdi.id], { VM: userData.vm.id })
+        return vbd !== undefined && vbd.attached
+      },
+      handler: deleteVbd,
+      icon: 'vdi-forget',
+      label: _('vdiForget'),
+      level: 'danger',
+    },
+    {
+      disabled: (vdi, userData) => {
+        const vbd = find(userData.vbdsByVdi[vdi.id], { VM: userData.vm.id })
+        return vbd !== undefined && vbd.attached
+      },
+      handler: deleteVdi,
+      icon: 'vdi-remove',
+      label: _('vdiRemove'),
+      level: 'danger',
+    },
+  ]
   render () {
     const { srs, vbds, vdis, vm } = this.props
 
     const { attachDisk, bootOrder, newDisk } = this.state
 
+    const _getVbdsByVdi = createSelector(
+      () => vdis,
+      vdis => {
+        const vbdsByVdi = []
+        forEach(vdis, vdi => {
+          vbdsByVdi[vdi.id] = map(vdi.$VBDs, id => find(vbds, { id }))
```

Comment:
You only need the VBD that is attached to the VM. And this will allow you to remove all the `_.find`s in the `itemRenderer`s:
```js
vbdByVdi[vdi.id] = find(vbds, { VDI: vdi.id, VM: vm.id })
````